src/build/__main__.py

```diff
@@ -239,7 +240,7 @@ def build_package_via_sdist(
     built: list[str] = []
     if distributions:
         # extract sdist
-        with tarfile.open(sdist) as t:
+        with TarFile.open(sdist) as t:
```

Comment:
You don't need this new class.From https://docs.python.org/3/library/tarfile.html:

```
t.extraction_filter = getattr(tarfile, 'data_filter', (lambda member, path: member))
```

on the line before the `extractall()` call.

Reply:
Doing it that way at call site is a little unsightly and it's not reusable. I'd prefer to keep the subclass.